mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLOps.td

```diff
@@ -59,6 +59,13 @@ class SYCLMethodOpInterfaceImpl<
   }];
 }
 
+class SYCLConstructor<string name, list<Trait> traits = []>
```

Comment:
We have the existing SYCLConstructorOp, is this new operation intended to eventually replace the existing way to construct SYCL objects ?  Having both would not be ideal. 

We would have to modify existing transformations and analysis. So there is an overhead to take into consideration. What are the main benefits in your opinion of this new approach ?

Reply:
The lowering of SYCLConstructorOp is to call the mangled name provided as one of the operation attributes, so transformations that want to construct an id, cannot insert `SYCLConstructorOp`, instead transformations need to perform a chain of operations, `alloca, id.get, load, store, ...` which makes debugging and lit test complicated.

Reply:
Ideally, each `sycl` type would have a `sycl.ty.constructor` operation to avoid having to lower to a function call. This introduces `sycl.id.constructor`. We are not going to modify codegen, this is an auxiliary operation to be used by transformations.

We could add canonicalization patterns so that this operation gets introduced in the pipeline, though, but I'd argue that's not critical.